components/elements/modals/BaseModal/BaseModal.test.ts

```diff
@@ -1,44 +1,33 @@
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-expect-error
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-import { setup, fetch, $fetch } from '@nuxt/test-utils-edge'
+import { describe, expect, test } from 'vitest'
+import { mount } from '@vue/test-utils'
 
-import { describe, test as it } from 'vitest'
 import BaseModal from './BaseModal.vue'
-import { fireEvent, stubbedRender } from '@/testUtils'
 
-describe('<BaseModal />', async () => {
-  await setup({})
-
-  it('should render without crashing', () => {
-    const { unmount } = stubbedRender(BaseModal)
-
-    unmount()
+describe('<BaseModal />', () => {
+  const BaseModalWrapper = mount(BaseModal, {
+    slots: { default: 'Modal content' },
   })
 
-  it('renders the correct <slot />', () => {
-    const { getByText } = stubbedRender(BaseModal, {
-      slots: { default: 'Modal content' },
-    })
+  test('should render without crashing', () => {
+    expect(BaseModalWrapper.isVisible()).toBe(true)
+  })
 
-    expect(getByText('Modal content')).toBeInTheDocument()
+  test('renders the correct <slot />', () => {
+    expect(BaseModalWrapper.html()).toContain('Modal content')
   })
 
   describe('when the close event is emitted', () => {
     test('when the close button is clicked', async () => {
-      const { emitted, getByTestId } = stubbedRender(BaseModal)
-
-      await fireEvent.click(getByTestId('modal-close-button'))
+      // fireEvent.click(getByTestId('modal-close-button'))
+      await BaseModalWrapper.find('button').trigger('click')
 
-      expect(emitted().close).toBeTruthy()
+      expect(BaseModalWrapper.emitted().close).toHaveLength(1)
```

Comment:
What happens if we _do_ have more than one `close` event?

Reply:
then it would have multiple emits. And would adjust the test as such. if right now given we are clicking once if it isn't emitting 1 time then that's unexpected behavior and needs to be addressed.